File: kalkylatorer/birth.js
function calculate() {
    let EDD = document.getElementById("EDD").value;
    let fd = document.getElementById("fd").value;
    let preg = document.getElementById("preg").value;
    const divisor = 86400000
    let normal = 280;
    let calcEDD = EDD.split(" ").map(str => Number(str));
    let calcFD = fd.split(" ").map(str => Number(str));
    let calcPreg = preg.trim().split("+").map(str => Number(str));
    if (EDD && fd) {
        let date1 = new Date(calcFD[0], calcFD[1] - 1, calcFD[2]);
        let date2 = new Date(calcEDD[0], calcEDD[1] - 1, calcEDD[2]);
        let delta = (date1.getTime() - date2.getTime()) / divisor;
        document.getElementById("output").value = `${Math.floor((normal + delta) / 7)} + ${(normal + delta) % 7}`;
    } else if (EDD && preg) {
        let offset = ((calcPreg[0] * 7) + calcPreg[1]) - normal;
        let EDD = new Date(calcEDD[0], calcEDD[1] - 1, calcEDD[2]);
        console.log(EDD, offset, calcPreg);
        document.getElementById("output").value = `${new Date(EDD.setDate(EDD.getDate() + offset)).toDateString()}`;
    } // else if (fd != -1) {
    //     offset = normal - ((pregnancy[0] * 7) + pregnancy[1])
    //     fd = datetime.date(fd[0], fd[1], fd[2])
    //     print(f 'Calculated at: {fd + datetime.timedelta(offset)}')
    // }
}

File: kalkylatorer/birth.py
import PySimpleGUI as sg
import datetime

layout = [
    [
        sg.Text("EDD (yyyy mm dd)"),
        sg.In(enable_events=True, key="EDD"),
    ],
    [
        sg.Text("fd (yyyy mm dd)"),
        sg.In(enable_events=True, key="fd"),
    ],
    [
        sg.Text("pregnancy (weeks days)"),
        sg.In(enable_events=True, key="pregnancy"),
    ],
    [
        sg.Button("calculate")
    ],
    [
        sg.Output(key="out")
    ]
]

window = sg.Window("Test", layout)

while True:
    event, values = window.read()
    # End program if user closes window or
    # presses the OK button
    if event == "calculate":
        normal = 279
        EDD = [int(i) for i in values["EDD"].split()]
        fd = [int(i) for i in values["fd"].split()]
        pregnancy = [int(i) for i in values["pregnancy"].split()]
        window.FindElement('out').Update('')
        if EDD and fd:
            delta = datetime.date(fd[0], fd[1], fd[2]) - datetime.date(EDD[0], EDD[1], EDD[2])
            print(f'{(normal + delta.days) // 7} + {(normal + delta.days) % 7}')
        elif EDD:
            offset = ((pregnancy[0] * 7) + pregnancy[1]) - normal
            EDD = datetime.date(EDD[0], EDD[1], EDD[2])
            print(f'Born at: {EDD + datetime.timedelta(offset)}')

        elif fd:
            offset = normal - ((pregnancy[0] * 7) + pregnancy[1])
            fd = datetime.date(fd[0], fd[1], fd[2])
            print(f'Calculated at: {fd + datetime.timedelta(offset)}')

        else:
            print("Non acceptable input")

    if event == sg.WIN_CLOSED:
        break

window.close()
